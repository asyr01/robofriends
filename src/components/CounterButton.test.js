import { shallow, mount, render } from 'enzyme';
import CounterButton from './CounterButton';
import React from 'react';
import renderer from 'react-test-renderer';

it('expect to render CounterButton Component', () => {
  const mockColor = 'red';
  const tree = renderer.create(<CounterButton color={mockColor} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('correctly increments the counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 3 });
  wrapper.find('[id="counter"]').simulate('keypress');
  expect(wrapper.state()).toEqual({ count: 3 });
  expect(wrapper.props().color).toEqual('red');

  const tree = renderer.create(<CounterButton color={mockColor} />).toJSON();
  expect(tree).toMatchSnapshot();
});
