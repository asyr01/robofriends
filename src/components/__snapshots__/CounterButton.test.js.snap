// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly increments the counter 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;

exports[`expect to render CounterButton Component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
